### PR TITLE
chore: Add more information about classpath and compiler options when severe error happens

### PR DESCRIPTION
**Pull request by tgodzik**
Related to https://github.com/scalameta/metals/issues/5272

### CODE REVIEW

File: mtags-shared/src/main/scala/scala/meta/internal/pc/CompilerAccess.scala [hunk 1]
```diff
@@ -28,8 +28,18 @@ abstract class CompilerAccess[Reporter, Compiler](
     config: PresentationCompilerConfig,
     sh: Option[ScheduledExecutorService],
     newCompiler: () => CompilerWrapper[Reporter, Compiler],
-    shouldResetJobQueue: Boolean
+    shouldResetJobQueue: Boolean,
+    additionalReportingData: () => String
 )(implicit ec: ExecutionContextExecutor, rc: ReportContext) {
+
+  def this(
```
Comment (kasiaMarek) [inline mtags-shared/src/main/scala/scala/meta/internal/pc/CompilerAccess.scala:35]: Is there a difference between this and giving a default value to the `additionalReportingData` arg?

File: mtags/src/main/scala-3/scala/meta/internal/pc/ScalaPresentationCompiler.scala [hunk 2]
```diff
@@ -64,6 +64,12 @@ case class ScalaPresentationCompiler(
       config,
       sh,
       () => new Scala3CompilerWrapper(newDriver),
+      () => s"""|Scala version: $scalaVersion
+                |Classpath:
+                |${classpath.mkString(", ")}
```
Comment (kasiaMarek) [inline mtags/src/main/scala-3/scala/meta/internal/pc/ScalaPresentationCompiler.scala:69]: Don't you want to have that `missing`, `exists` info for classpath here too?

**Review by kasiaMarek — APPROVED** [review]

File: mtags-shared/src/main/scala/scala/meta/internal/pc/CompilerAccess.scala (hunk 1 above)
Comment (tgodzik) [reply mtags-shared/src/main/scala/scala/meta/internal/pc/CompilerAccess.scala:35]: It seems there is for the JVM, it fails in with the default argument.

File: mtags/src/main/scala-3/scala/meta/internal/pc/ScalaPresentationCompiler.scala (hunk 2 above)
Comment (tgodzik) [reply mtags/src/main/scala-3/scala/meta/internal/pc/ScalaPresentationCompiler.scala]: good catchm added.

**Pull request merged**